tools/apiview/parsers/csharp-api-parser/CSharpAPIParserTests/CodeFileTests.cs

```diff
@@ -7,33 +7,163 @@
 using Newtonsoft.Json.Linq;
 using System.Text.Json.Serialization;
 using APIView.Model.V2;
+using Microsoft.CodeAnalysis;
+using NuGet.ContentModel;
 
 
 namespace CSharpAPIParserTests
 {
     public class CodeFileTests
     {
-        private readonly CodeFile codeFile;
-        public Assembly assembly { get; set; }
+        private readonly CodeFile templateCodeFile;
+        private Assembly templateAssembly { get; set; }
+        private readonly CodeFile storageCodeFile;
+        public Assembly storageAssembly { get; set; }
+        private readonly CodeFile coreCodeFile;
+        public Assembly coreAssembly { get; set; }
 
         public CodeFileTests()
         {
-            assembly = Assembly.Load("Azure.Template");
-            var dllStream = assembly.GetFile("Azure.Template.dll");
+            templateAssembly = Assembly.Load("Azure.Template");
+            var dllStream = templateAssembly.GetFile("Azure.Template.dll");
             var assemblySymbol = CompilationFactory.GetCompilation(dllStream, null);
-            this.codeFile = new CSharpAPIParser.TreeToken.CodeFileBuilder().Build(assemblySymbol, true, null);
+            this.templateCodeFile = new CSharpAPIParser.TreeToken.CodeFileBuilder().Build(assemblySymbol, true, null);
+
+            this.storageAssembly = Assembly.Load("Azure.Storage.Blobs");
+            dllStream = storageAssembly.GetFile("Azure.Storage.Blobs.dll");
+            assemblySymbol = CompilationFactory.GetCompilation(dllStream, null);
+            this.storageCodeFile = new CSharpAPIParser.TreeToken.CodeFileBuilder().Build(assemblySymbol, true, null);
+
+            this.coreAssembly = Assembly.Load("Azure.Core");
+            dllStream = coreAssembly.GetFile("Azure.Core.dll");
+            assemblySymbol = CompilationFactory.GetCompilation(dllStream, null);
+            this.coreCodeFile = new CSharpAPIParser.TreeToken.CodeFileBuilder().Build(assemblySymbol, true, null);
         }
 
         [Fact]
         public void TestPackageName()
         {
-            Assert.Equal("Azure.Template", codeFile.PackageName);
+            Assert.Equal("Azure.Template", templateCodeFile.PackageName);
```

Comment:
for this and the other basic package metadata validation tests, you might want to add a parameterized `Theory` to cover all the codefiles in this class:

```c#
[Theory]
[MemberData(nameof(CodeFileNames))]
public void TestPackageName(CodeFile codeFile, string expectedName)
{
    Assert.Equal(expectedName, codeFile.PackageName);
}

public static IEnumerable<object[]> CodeFileNames => new List<object[]>
{
    new object[] { templateCodeFile, "Azure.Template" },
    new object[] { storageCodeFile , "Azure.Storage.Blobs" },
    new object[] { coreCodeFile, "Azure.Core" },
};
```